apps/src/codebridge/FileBrowser/hooks/useHandleFileUpload.ts

```diff
@@ -58,10 +55,7 @@ export const useHandleFileUpload = (
         return;
       }
 
-      const fileId = getNextFileId(Object.values(projectFiles));
-
```

Comment:
per the description, this is just the re-tooling so that `getNextFileId` is only called internally in the reducer.

It might also be appropriate to refactor away the `getNextFileId` and `getNextFolderId` functions, or at least make them private inside the reducer itself, but I left them as is and external for now.

Reply:
I think I missed this case when reworking validation, but I think we need to reuse [this logic](https://github.com/code-dot-org/code-dot-org/pull/61463/files#diff-e6c06fe9e120a0fe01bc04e351eef193fa9d10350d231ff799b7396ad1dc90e7R521-R526) so an uploaded file doesn't overwrite the id of a validation file. We could do this separately since this is a pre-existing issue, but I don't think it will work to move it into the reducer since the reducer doesn't know about the validation file.

Reply:
Ouch. That precludes the use of multiple file uploads, at least without doing some serious hoop jumping around asynchronous calling of multiple `newFile`s in a row, but that'd be tough to implement and keep in sync. I don't recommend it.

Okay. Do we have any additional information about the validation file ids, and/or could we standardize them?

For example, if all of the ids on validation files are known to be "0", then there's no problem - `getNextFileId` would always return at least `1`. Also, since the ids are strings and not numbers, we _should_ be able to give those support files Ids outside of the numeric range. So a literal id of `validation`, for instance (the only gotcha would be if there's code anywhere that's assuming the IDs can be cast to numbers, but I don't _think_ there is. Nothing should be making any sort of ordering assumptions in that manner).

Or we could do something like reserve odd numbers for internal support files and even numbers for user created, then re-tool `getNextFileId` to increment by `2`.

Is there any consistency in the validationFile ids right now? It probably wouldn't be worth it to do a large scale migration of the existing files to a standard ID.

Although...the `validationFile` is loaded separately from `lab.levelProperties.validationFile`, right? We could potentially re-write that object (maybe even on the fly?) to just give it a new id (`validation`?) so it's a standardized value outside of the scope and can't be stomped on?

I'm not sure what other good options there would be atm. `getNextFileId` really _should_ be internal to the reducer, and it was a leaky interface to have it on the outside previously, I just couldn't figure out a way to get rid of it initially.

Reply:
Hrm. I suppose alternatively, we could give the `validationFile` as an arg to `newFile` as well, and have that incorporate into how `getNextFileId` calculates the value. That way we keep still keep it internal, at expense of a messier interface. 

Lemme look into that a little more, it may be the easiest option.

Reply:
> Although...the validationFile is loaded separately from lab.levelProperties.validationFile, right? We could potentially re-write that object (maybe even on the fly?) to just give it a new id (validation?) so it's a standardized value outside of the scope and can't be stomped on?

Yes, we could standardize the id on load; I considered doing that initially but at the time it seemed more trouble than it was worth (because we have to make sure the other references to the id also get updated correctly)

Reply:
I went with the second approach. I retooled `getNextFileId` to optionally accept the `validationFileId` as the second argument, and then `validationFileId` is handed through to `getNextFileId`.

If it's passed through, then it'll also be handed into the max list when the next id is calculated, and if it's not it'll just default to "0" as a no-op.

I think that'll cover it, right? `newFile` is only called in `openNewFilePrompt` and `handleFileUpload`

mods are in this commit: https://github.com/code-dot-org/code-dot-org/pull/61921/commits/5d8862cd4f62d71da23f830b39835a356ea8786e